mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -287,6 +292,28 @@ def enclosure(self):
             LOG.error(simple_trace(traceback.format_stack()))
             raise Exception('Accessed MycroftSkill.enclosure in __init__')
 
+    @property
+    def file_system(self):
+        """ Filesystem access to skill specific folder.
+
+        See mycroft.filesystem for details.
+        """
+        if not self._file_system and self.skill_id:
+            self._file_system = FileSystemAccess(join('skills', self.skill_id))
+        if self._file_system:
+            return self._file_system
+        else:
+            LOG.error('Skill not fully initialized. Move code ' +
+                      'from  __init__() to initialize() to correct this.')
+            LOG.error(simple_trace(traceback.format_stack()))
+            raise Exception('Accessed MycroftSkill.file_system in __init__')
+
+    @file_system.setter
+    def file_system(self, fs):
+        """Provided mainly for backwards compatibility with derivative MycroftSkill classes
+        Skills are advised against redefining the file system directory"""
```

Comment:
Maybe log a warning if this is intended to be deprecated?

Reply:
its not deprecated but it should be considered read only

i made it a property but previously it was a variable, i allow assigning to it to keep the api compatible in case something like neon is overriding it

Reply:
the reason this changed is that its easier to account for skill_id this way, otherwise i would need some _init_filesystem method, but this can cause issues depending on what a skill does between init and initialize

i think homescreen skill runs (used to?) into this when handling wallpapers, it accesses file system in init,  so doing it this way indicates that breaking change

Reply:
Is the setter going to remain, or is it planned to deprecate it later? If this is sticking around then this is fine

Reply:
it will stay, not planned for removal